cli/cmd/download_untar_lz4.go

```diff
@@ -0,0 +1,104 @@
+package cmd
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+func downloadUntarLz4(ctx context.Context, srcURL string, outputDir string) error {
+	// Build an HTTP GET request with an injected context.
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, srcURL, nil)
+	if err != nil {
+		return errors.Wrap(err, "build GET request")
+	}
+
+	// Send the request.
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "send request")
+	}
+	defer resp.Body.Close()
+
+	// Check if the HTTP status is OK.
+	if resp.StatusCode != http.StatusOK {
+		return errors.Wrap(err, "download file", "status_code", resp.StatusCode)
+	}
+
+	if err := untarLz4(ctx, resp.Body, outputDir); err != nil {
+		return errors.Wrap(err, "decompress tar lz4", "url", srcURL)
+	}
+
+	return nil
+}
+
+func untarLz4(ctx context.Context, stream io.Reader, outputDir string) error {
+	// Create an LZ4 reader.
+	lz4Reader := lz4.NewReader(stream)
+
+	// Decompress into memory or stream directly.
+	var buf bytes.Buffer
+	if _, err := io.Copy(&buf, lz4Reader); err != nil {
+		return errors.Wrap(err, "decompress lz4")
+	}
+
```

Comment:
this will load the whole thing into memory, rather do as I showed `tar.NewReader(lz4.NewReader(stream))`

Reply:
Done [16cac9c](https://github.com/omni-network/omni/pull/2782/commits/16cac9c22a8fbd0732580f4a6755d73deadc88d4)